### PR TITLE
fix: fix wasmer build error

### PR DESCRIPTION
**Pull request by wacban**
Running `cargo test` in the nearcore directory fails with build error as shown below. This PR fixes this issue by making the relevant code only compiled under the right conditions e.i. when the wasmer-runtime feature is enabled. 

Tested with:
- `cargo build`
- `cargo test --no-run` 
- TODO 
    - How to test it with the wasmer feature enabled? 
    - I tried `cargo build --features force_wasmer` but it fails with:
```
error: Wasmer only supports x86_64, but a force_wasmer* feature was passed to near-vm-runner
  --> runtime/near-vm-runner/src/vm_kind.rs:28:9
```

The original error is:
```
~/near/nearcore % cargo test --no-run

error[E0433]: failed to resolve: use of undeclared crate or module `wasmer_types`
  --> runtime/near-vm-runner/src/tests.rs:74:23
   |
74 |     const PAGE: u64 = wasmer_types::WASM_PAGE_SIZE as u64;
   |                       ^^^^^^^^^^^^ use of undeclared crate or module `wasmer_types`

error[E0080]: evaluation of constant value failed
  --> runtime/near-vm-runner/src/tests.rs:78:19
   |
78 |         buf: [u8; PAGE as usize + 1],
   |                   ^^^^ referenced constant has errors

```

### CODE REVIEW

**Comment by akhi3030** [comment]
@nagisa: maybe you can take a look at this.

File: runtime/near-vm-runner/src/tests.rs [hunk 1]
```diff
@@ -70,6 +72,7 @@ fn prepaid_loading_gas(bytes: usize) -> u64 {
 /// memory must be configured for indices `0..WASM_PAGE_SIZE` to be valid.
 ///
 /// Panics if any of the tests fails.
+#[cfg(all(feature = "wasmer-runtime", target_arch = "x86_64"))]
```
Comment (nagisa) [inline runtime/near-vm-runner/src/tests.rs:75]: This test function is intended to be applicable for all runtimes we implement. If the `PAGE` constant in there is causing problems, I think it'd make sense to hardcode its value (`64 * 1024` IIRC)

**Review by nagisa — CHANGES_REQUESTED** [review]

File: runtime/near-vm-runner/src/tests.rs (hunk 1 above)
Comment (wacban) [reply runtime/near-vm-runner/src/tests.rs:75]: It is the PAGE constant, I set it to the value from: 
https://docs.rs/wasmer/latest/wasmer/constant.WASM_PAGE_SIZE.html .
In cargo build this code generates unused code warnings so I added allow(dead_code). It solves the problem but I feel like if we could find the right feature/ target_arch that would be a cleaner.

Comment (nagisa) [reply runtime/near-vm-runner/src/tests.rs]: The webassembly page size is specified by the webassembly specification, so there is really no strong reason to depend on any particular runtime to provide this value over specifying one in-line.

**Pull request merged**